src/ast/mod.rs

```diff
@@ -55,16 +55,24 @@ pub(crate) enum Ruleset {
 pub(crate) const DEFAULT_FILENAME: &str = "<unnamed.egg>";
 pub(crate) const DUMMY_FILENAME: &str = "<internal.egg>";
 
+#[derive(Clone, Debug, PartialEq, Eq, Hash)]
+pub struct SrcFile {
+    pub name: String,
+    pub contents: Option<String>,
+}
+
 /// A [`Span`] contains the file name and a pair of offsets representing the start and the end.
 #[derive(Clone, Debug, PartialEq, Eq, Hash)]
-pub struct Span(pub Symbol, pub usize, pub usize);
+pub struct Span(pub Arc<SrcFile>, pub usize, pub usize);
 
 lazy_static! {
-    pub(crate) static ref DUMMY_SPAN: Span = Span(Symbol::from(DUMMY_FILENAME), 0, 0);
+    pub(crate) static ref DUMMY_FILE: Arc<SrcFile> = Arc::new(SrcFile {
+        name: DUMMY_FILENAME.to_string(),
+        contents: None
+    });
+    pub(crate) static ref DUMMY_SPAN: Span = Span(DUMMY_FILE.clone(), 0, 0);
```

Comment:
What if Span was an enum (or use an `Option<Span>`)?  I was just thinking if error reporting then we could differentiate between having no line number versus outputting a dummy one.

I don't think it's a big deal though either way.